File: src/main/java/com/nbteam/killer/helper/repository/GameBookRepository.java
package com.nbteam.killer.helper.repository;

import com.nbteam.killer.helper.domain.GameBook;

import java.util.List;

public interface GameBookRepository {

    /**
     * 通过玩家数量随机取本
     *
     * @param playerTotal
     * @return
     * @throws Exception
     */
    GameBook getRandom(Integer playerTotal) throws Exception;

    /**
     * 通过板子id创建板子
     *
     * @param id
     * @return
     */
    GameBook getById(String id) throws Exception;

    /**
     * 通过板子id创建板子
     *
     * @param id
     * @param isBdModal
     * @return
     */
    GameBook getById(String id, Boolean isBdModal) throws Exception;

    /**
     * 获取板子集合
     *
     * @return
     */
    List<GameBook> list();
}

File: killer-helper-base/src/main/kotlin/com/nbteam/killer/helper/base/repository/GameBookRepository.kt
package com.nbteam.killer.helper.repository

import com.nbteam.killer.helper.base.domain.GameBook

interface GameBookRepository {
    /**
     * 通过玩家数量随机取本
     *
     * @param playerTotal
     * @return
     * @throws Exception
     */
    @Throws(Exception::class)
    fun getRandom(playerTotal: Int?): GameBook?

    /**
     * 通过板子id创建板子
     *
     * @param id
     * @return
     */
    @Throws(Exception::class)
    fun getById(id: String?): GameBook?

    /**
     * 通过板子id创建板子
     *
     * @param id
     * @param isBdModal
     * @return
     */
    @Throws(Exception::class)
    fun getById(id: String?, isBdModal: Boolean?): GameBook?

    /**
     * 获取板子集合
     *
     * @return
     */
    fun list(): List<GameBook?>?
}